### ISSUE TITLE
migrating from withDataProvider to the hook

### ISSUE THREAD

**Issue by ybrodsky** [marmelab/react-admin#4955]
In version 2.x when you used the hoc `withDataProvider` and you used the `dataProvider` injected into the props, you could pass some extra options like `onSuccess`, `onFailure` and additionally using that dataProvider made sure all the proper redux actions were called. So you could see the loading indicator and whatnot.

Now with the new hook, you can no longer pass on those options. So you are forced to use also `useNotify`, `useRefresh`, etc.
I noticed that also no redux actions are being called.... so no loading indicator.

So basically my question is how to use the dataProvider and have it show the loading indicator, but it could be well be paraphrased as have it run all the redux actions that are normally ran when you use the default components like `List` or `View`

**Comment by djhi** [comment]
Hi, and thanks for your question. As explained in the [react-admin contributing guide](https://github.com/marmelab/react-admin/blob/master/README.md#contributing), the right place to ask a "How To" question, get usage advice, or troubleshoot your own code, is [StackOverFlow](http://stackoverflow.com/questions/tagged/react-admin).

This makes your question easy to find by the core team, and the developer community. Unlike Github, StackOverFlow has great SEO, gamification, voting, and reputation. That's why we chose it, and decided to keep GitHub issues only for bugs and feature requests.

So I'm closing this issue, and inviting you to ask your question at:

http://stackoverflow.com/questions/tagged/react-admin

And once you get a response, please continue to hang out on the react-admin channel in StackOverflow. That way, you can help newcomers and share your expertise!

**Comment by ybrodsky** [comment]
With all due respect, 50% of the times you google something you end up finding the solution in a github issue. They googleable the same way stackoverflow is.
I get you don't want to flood issues with basic questions but this is not a "how" question. Unless you can point me to a place of the documentation I missed in which case I will admit being an idiot and apologise.

I will go make the stackoverflow question, thanks.

**Comment by fzaninotto** [comment]
You mean like https://marmelab.com/react-admin/Actions.html#handling-side-effects-in-usedataprovider?

**Comment by ybrodsky** [comment]
No. I was migrating. 
I answered my own question on stackoverflow. Turns out you have to migrate everything, even legacy stuff that still 'worked"
